pkg/ocm/ocm.go

```diff
@@ -146,37 +148,49 @@ func (c *Controller) checkSecret(ocmData *ScaResponse) error {
 		return err
 	}
 
+	if scaSec.Type == v1.SecretTypeTLS {
+		err = c.coreClient.Secrets(targetNamespaceName).Delete(c.ctx, scaSec.GetName(), metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+		err = c.createSecret(ocmData)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
```

Comment:
Is it possible to do the TLS check before checking the existence of secret (line 140)?  That way you don't need to duplicate the code at lines 156-160.  Then the logic is:

If TLS secret exists, delete it.  
If no secret, create one.

Reply:
Hmm... I don't think so (or I don't know how to do that). The `CoreV1Interface` doesn't allow you to get or list secrets of specific type (pls correct me if I am wrong)

Reply:
This is minor, and I am not familiar with the code, but wouldn't this simplify the logic?

\`func (c \*Controller) checkSecret(ocmData \*ScaResponse) error {  
   scaSec, err := c.coreClient.Secrets(targetNamespaceName).Get(c.ctx, secretName, metav1.GetOptions{})  
   if err != nil {  
       return err  
   }

   if scaSec.Type == v1.SecretTypeTLS {  
       delErr := c.coreClient.Secrets(targetNamespaceName).Delete(c.ctx, scaSec.GetName(), metav1.DeleteOptions{})  
       if delErr != nil {  
           return delErr  
       }  
   }

   // if the secret doesn't exist then create one  
   if errors.IsNotFound(err) {  
       return c.createSecret(ocmData)  
   }\`

Reply:
if the secret is deleted then I have to created. Other than that your snippet is moving the ` if errors.IsNotFound(err)` few lines down, which is only "cosmetic" change IMO (and I think it's better to handle the error immediately).

It's true that I can return `c.createSecret(ocmData)` and save few lines with the error checking. I can update it if you want...but again a minor change IMO.

Reply:
Okay, will let this go.

Reply:
OK thanks a lot.